### ISSUE TITLE
some csv function "Alternative invocations" examples do not work

### ISSUE THREAD

**Issue by lpenninger** [makamaka/Text-CSV#46]
In the Alternative invocations section of the csv function documentation, the second and third examples do not work:

```
my $csv = Text::CSV->new ();
my $aoa = $csv->csv (in => "file.csv");

my $csv = Text::CSV->new ({ sep_char => ";" });
my $aoh = $csv->csv (in => "file.csv", bom => 1);
```

When each of these is run, you get the error: 
> Odd number of elements in hash assignment at /opt/ActivePerl-5.18/site/lib/Text/CSV_XS.pm line 1120.

This is because the Text::CSV object is passed as the first argument, and it does not pass the check at the beginning of Text::CSV_XS::_csv_attr:
`      my %attr = (@_ == 1 && ref $_[0] eq "HASH" ? %{$_[0]} : @_) or croak;`

which causes @_ to be returned as an odd number of elements.

Versions: Text-CSV_XS-1.41, Text-CSV-2.00

**Comment by xtaran** [comment]
Just ran into this with Text::CSV 2.01 and Text::CSV::XS 1.48:

```
Odd number of elements in hash assignment at /usr/lib/x86_64-linux-gnu/perl5/5.34/Text/CSV_XS.pm line 1171.
usage: my $aoa = csv (in => $file); at …
```

It also happens if I enforce the pure Perl implementation with `PERL_TEXT_CSV=Text::CSV_XS`:

```
usage: my $aoa = csv (in => $file); at …
```

My code was more or less this:

```perl
#!/usr/bin/perl

use strict;
use warnings;
use 5.010;
use Text::CSV;
use utf8;

my $csv = Text::CSV->new();
my $data = $csv->csv(in => $ARGV[0] ,
                     headers => 'auto',
                     encoding => 'UTF-8');
```

**Comment by charsbar** [comment]
Fixed by https://github.com/makamaka/Text-CSV/commit/81995e48967d5a90f02127e1f38b54ad0ea1e450 and shipped Text::CSV 2.02. Thank you for the patience.
